src/types/timestamp.rs

```diff
@@ -1584,6 +1634,117 @@ mod timestamp_tests {
         Ok(())
     }
 
+    #[test]
+    fn test_timestamp_year() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.year(), 2021);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_month() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.month(), 2);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_day() -> IonResult<()> {
```

Comment:
We need some tests for the `year`/`month`/`day` accessors where the `Timestamp` has high precision and an offset.